application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,196 @@
+
+        configModMailChannelPattern = config.getModMailChannelPattern();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        long userID = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userID, event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+
+        String reportedMessage = event.getTarget().getContentRaw();
+        String reportedMessageUrl = event.getTarget().getJumpUrl();
+
+        TextInput body = TextInput
+            .create(REPORT_REASON_INPUT_ID, "Anonymous report to the moderators",
+                    TextInputStyle.PARAGRAPH)
+            .setPlaceholder("What is wrong with the message, why do you want to report it?")
+            .setRequiredRange(3, 200)
+            .build();
+
+        Modal modal =
+                Modal.create(generateComponentId(reportedMessage, reportedMessageUrl), "Report")
+                    .addActionRow(body)
+                    .build();
+
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog =
+                handleRequireModMailChannel(event, event.getJDA(), guildID);
+
+        if (modMailAuditLog.isEmpty()) {
+            return;
+        }
+
+        sendModMessage(event, args, modMailAuditLog.orElseThrow());
+    }
+
+    private Optional<TextChannel> handleRequireModMailChannel(ModalInteractionEvent event, JDA jda,
+            long guildID) {
+        Optional<TextChannel> modMailAuditLog = jda.getGuildById(guildID)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+        if (modMailAuditLog.isEmpty()) {
+            event
+                .reply("Sorry, there was an issue sending your report to the moderators. We are "
+                        + "investigating.")
+                .queue();
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+        }
+        return modMailAuditLog;
+    }
+
+    private boolean handleIsOnCooldown(long userID, MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(userID)) {
+            return false;
+        }
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
```

Comment:
"You can only report a message once per %s minutes"

Reply:
y even hard code "minutes"?

Reply:
certainly true, given that there is a `COOLDOWN_DURATION_UNIT` as well... probably not that easy to get it nice though.

i think i had some utility duplicated in 2 classes (somewhere around the moderation helper maybe?)

tbh, its okay for today. lets do that in another PR and align it elsewhere in other classes as well